src/main/kotlin/com/openlattice/entitysets/controllers/EntitySetsController.kt

```diff
@@ -402,6 +414,87 @@ constructor(
         return 1
     }
 
+    @Timed
+    @RequestMapping(
+            path = [ALL + ID_PATH + EXPIRATION_PATH],
+            method = [RequestMethod.PATCH],
+            produces = [MediaType.APPLICATION_JSON_VALUE]
+    )
+    override fun removeDataExpirationPolicy(
+            @PathVariable(ID) entitySetId: UUID
+    ): Int {
+        ensureOwnerAccess(AclKey(entitySetId))
+        edmManager.removeDataExpirationPolicy(entitySetId)
+
+        recordEvent(
+                AuditableEvent(
+                        getCurrentUserId(),
+                        AclKey(entitySetId),
+                        AuditEventType.UPDATE_ENTITY_SET,
+                        "Entity set data expiration policy removed through EntitySetApi.removeDataExpirationPolicy",
+                        Optional.empty(),
+                        ImmutableMap.of("entitySetId", entitySetId),
+                        OffsetDateTime.now(),
+                        Optional.empty()
+                )
+        )
+        return 1
+    }
+
+    @Timed
+    @RequestMapping(
+            path = [ALL + ID_PATH + EXPIRATION_PATH],
+            method = [RequestMethod.POST]
+    )
+    override fun getExpiringEntitiesFromEntitySet(
+            @PathVariable(ID) entitySetId: UUID,
+            @RequestBody dateTimeAsString: String): Set<UUID> {
+        ensureReadAccess(AclKey(entitySetId))
+        val dateTime = OffsetDateTime.parse(dateTimeAsString)
+        val es = getEntitySet(entitySetId)
+        check(es.hasExpirationPolicy()) { "Entity set ${es.name} does not have an expiration policy" }
+
+        val expirationPolicy = es.expiration
+        if (expirationPolicy != null) {
```

Comment:
I think the null check is not necessary at this point, because `es.hasExpirationPolicy()` checks if expiration policy is null or not (and throws exception if it is)

Reply:
oops! thanks for catching that